qa/acceptance-tests/src/test/java/io/camunda/it/client/RolesByMappingIntegrationTest.java

```diff
@@ -338,64 +243,272 @@ void shouldReturnNotFoundOnUnassigningRoleFromMappingIfRoleDoesNotExist() {
+    camundaClient.newAssignRoleToMappingCommand().roleId(roleId).mappingId(mappingB).send().join();
+
+    Awaitility.await("Mappings are sorted by claimValue")
+        .ignoreExceptionsInstanceOf(ProblemException.class)
+        .untilAsserted(
+            () -> {
+              final var result =
+                  camundaClient
+                      .newMappingsByRoleSearchRequest(roleId)
+                      .sort(s -> s.claimValue().desc())
+                      .send()
+                      .join();
+
+              assertThat(result.items())
+                  .extracting(Mapping::getClaimValue)
+                  .containsExactly(valueB, valueA);
+            });
+  }
+
+  @Test
+  void shouldFilterByMappingName() {
+    final var roleId = Strings.newRandomValidIdentityId();
+    final var mappingId = Strings.newRandomValidIdentityId();
+    final var name = "filter-name-" + Strings.newRandomValidIdentityId();
+
+    createRole(roleId, "FilterRole", "desc");
+    createMapping(
+        mappingId, name, Strings.newRandomValidIdentityId(), Strings.newRandomValidIdentityId());
+    camundaClient.newAssignRoleToMappingCommand().roleId(roleId).mappingId(mappingId).send().join();
+
+    Awaitility.await("Mapping is filtered by name")
+        .ignoreExceptionsInstanceOf(ProblemException.class)
+        .untilAsserted(
+            () -> {
+              final var result =
+                  camundaClient
+                      .newMappingsByRoleSearchRequest(roleId)
+                      .filter(f -> f.name(name))
+                      .send()
+                      .join();
+              assertThat(result.items())
+                  .singleElement()
+                  .extracting(Mapping::getName)
+                  .isEqualTo(name);
+            });
+  }
+
+  @Test
+  void shouldFilterByClaimName() {
+    final var roleId = Strings.newRandomValidIdentityId();
+    final var mappingId = Strings.newRandomValidIdentityId();
+    final var claimName = "filter-claimName-" + Strings.newRandomValidIdentityId();
+
+    createRole(roleId, "FilterRole", "desc");
+    createMapping(
+        mappingId,
+        Strings.newRandomValidIdentityId(),
+        claimName,
+        Strings.newRandomValidIdentityId());
+    camundaClient.newAssignRoleToMappingCommand().roleId(roleId).mappingId(mappingId).send().join();
+
+    Awaitility.await("Mapping is filtered by claimName")
+        .ignoreExceptionsInstanceOf(ProblemException.class)
+        .untilAsserted(
+            () -> {
+              final var result =
+                  camundaClient
+                      .newMappingsByRoleSearchRequest(roleId)
+                      .filter(f -> f.claimName(claimName))
+                      .send()
+                      .join();
+              assertThat(result.items())
+                  .singleElement()
+                  .extracting(Mapping::getClaimName)
+                  .isEqualTo(claimName);
+            });
+  }
```

Comment:
🔧 I think it makes sense to ensure there is at least 2 mappings assigned to the role when testing filters. This way we know that the one mapping with the wrong value actually gets filtered out.